### DISCUSSION TITLE
Kotlin curious?

### DISCUSSION THREAD

**Discussion by chris-hatton** [FAForever/downlords-faf-client#2975] [General]
Long time FAF player, client user and software dev here.

I notice the project is currently all in Java. While that's fine; I've also had a lot of good experiences with migrating Java code-bases to [Kotlin](https://kotlinlang.org/), making them easier to maintain: less error-prone, just less code-by-volume overall.  Would a PR to demo migrating of a small section of the code to Kotlin be welcome?  I understand it may not be, but if the Devs already here are curious, I could assist.

In case you're not aware, Kotlin is transparently interoperable with Java; there's no 'bridging' to speak of, you can include both languages side-by-side with no issues.  While the main benefit is probably one of maintenance, introducing Kotlin also provides a potential pathway to [Compose Multiplatform](https://www.jetbrains.com/lp/compose-multiplatform/), longer term, in case the project outgrows JavaFX.

**Comment by cdglitch** [comment]
I don't care what gets done, can we please just do something that doesnt break UI/UX for a change and actually make the client better?

**Comment by Brutus5000** [comment]
We use Kotlin elsewhere (https://github.com/FAForever/faf-telemetry-server, https://github.com/FAForever/faf-user-service, https://github.com/FAForever/faf-java-commons/tree/develop/faf-commons-lobby). From my experience with migrating large codebases I would stick to Java here. Java is improving a lot lately.

**Reply by chris-hatton** [reply]
Fair enough, just thought I'd offer.  Closing discussion.

**Reply by Brutus5000** [reply]
@chris-hatton If you're interested feel free to jump onto any of the projects mentioned above ;)
